_posts/blog/2016-10-28-durability-performance.md

```diff
@@ -0,0 +1,136 @@
+---
+title: "Durability Performance Implications"
+date: 2016-10-28 17:00:00 +0000
+author: Keith Turner
+---
+
+## Overview
+
+Accumulo stores recently written data in a sorted in memory map.  Before data is
+added to this map, it's written to an unsorted WAL (write ahead log).  In the
+case when a Tablet Server dies, the recently written data is recovered from the
+WAL.
+
+When data is written to Accumulo the following happens :
+
+ * Client sends a batch of mutations to a tablet server
+ * Tablet server does the following :
+   * Writes mutation to Tablet Servers WAL
+   * Sync or flush WAL
+   * Adds mutations to sorted in memory maps
```

Comment:
But isn't there still only one IMM, just partitioned by tablet? That was my distinction.

Also, "Tablet Servers' WAL" looking at this again :)